packages/node/src/ethereum/ethers/json-rpc-batch-provider.ts

```diff
@@ -112,8 +121,11 @@ export class JsonRpcBatchProvider extends JsonRpcProvider {
           batch.forEach((inflightRequest) => {
             inflightRequest.reject(error);
           });
+          this.failedBatchCount++;
+          this.adjustBatchSize();
```

Comment:
These always are called in pairs (or with`this.successfulBatchCount++;`). We should combine them together to make sure that they are always called together